lib/logical.ts

```diff
@@ -1,16 +1,19 @@
-module.exports = logicalTree;
-
-var pluck = require('./pluck');
-var walk = require('./walk');
-var unique = require('./unique');
-var path = require('path');
-var depTypes = require('./dep-types');
-var colour = require('ansicolors');
-var moduleToObject = require('snyk-module');
-var _ = require('./lodash');
-var format = require('util').format;
-var ext = colour.bgBlack(colour.green('extraneous'));
-var problems = [];
```

Comment:
this is global here, is there a test in place that can confirm this is still working as before?

Reply:
It's working better than before :D at least it's "async-safe" now.

Yes, there's a test that checks the presence of a "problem". But, in general, tests for this are very icky and don't look reliable. That's why I'd like the eyes of somebody who understands what this actually is supposed to be doing.